duktape.mk

```diff
@@ -0,0 +1,70 @@
+ifneq ($(PROCURSUS),1)
+$(error Use the main Makefile)
+endif
+
+SUBPROJECTS     += duktape
+DUKTAPE_VERSION := 2.6.0
+DEB_DUKTAPE_V   ?= $(DUKTAPE_VERSION)
+
+duktape-setup: setup
+	wget -q -nc -P $(BUILD_SOURCE) https://github.com/svaarala/duktape/releases/download/v$(DUKTAPE_VERSION)/duktape-$(DUKTAPE_VERSION).tar.xz
+	$(call EXTRACT_TAR,duktape-$(DUKTAPE_VERSION).tar.xz,duktape-$(DUKTAPE_VERSION),duktape)
+	$(SED) -i 's/gcc/cc/g' $(BUILD_WORK)/duktape/Makefile.cmdline
+	$(SED) -i 's|$$(CCLIBS)|$$(CCLIBS) $(LDFLAGS) $(CFLAGS)|g' $(BUILD_WORK)/duktape/Makefile.cmdline
+	mkdir -p $(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/{lib/pkgconfig,include}
+	cp $(BUILD_MISC)/duktape/duktape.pc $(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/pkgconfig/duktape.pc
+	$(SED) -i 's|prefix=|prefix=$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)|g' \
+	$(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/pkgconfig/duktape.pc
+	$(SED) -i 's/Version:/Version: $(DUKTAPE_VERSION)/g' \
+	$(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib/pkgconfig/duktape.pc
+
+
+ifneq ($(wildcard $(BUILD_WORK)/duktape/.build_complete),)
+duktape:
+	@echo "Using previously built duktape."
+else
+duktape: duktape-setup
+	cd $(BUILD_WORK)/duktape && python2 tools/configure.py --output-directory \
+	$(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/include -UDUK_USE_ES6_PROXY
+	+$(MAKE) -C $(BUILD_WORK)/duktape -f Makefile.sharedlibrary install \
+	CCOPTS="$(CFLAGS)" CCLIBS="$(LDFLAGS) -lm" \
+	INSTALL_PREFIX="$(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)"
+	+$(MAKE) -C $(BUILD_WORK)/duktape -f Makefile.cmdline
+	mkdir -p $(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
+	cp -a $(BUILD_WORK)/duktape/duk $(BUILD_STAGE)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin/duk
+	cp -a $(BUILD_STAGE)/duktape/* $(BUILD_BASE)
+	touch $(BUILD_WORK)/duktape/.build_complete
+endif
+
+duktape-package: duktape-stage
+	# duktape.mk Package Structure
+	rm -rf $(BUILD_DIST)/{libduktape206,duktape,duktape-dev}
+	mkdir -p $(BUILD_DIST)/duktape/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/bin
+	mkdir -p $(BUILD_DIST)/duktape-dev/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/{lib/pkgconfig,include,share/duktape}
+	mkdir -p $(BUILD_DIST)/libduktape206/$(MEMO_PREFIX)$(MEMO_SUB_PREFIX)/lib
```

Comment:
These can be done in one line.
```makefile
mkdir -p $(1) \
      $(2)
      $(3)
````

Reply:
I usually never do them all in one line

Reply:
Unless Hayden tells me to then I will

Reply:
Doing so reduces repetition; it's easier to read and looks less dumb.

Reply:
Do it in one line. Why run a command more than once

Reply:
Didn't realize you meant the mkdir lol

Reply:
fixed